container/import.bzl

```diff
@@ -40,9 +42,16 @@ load(
     _join_path = "join",
 )
 
-def _unzip_layer(ctx, zipped_layer):
-  unzipped_layer = _gunzip(ctx, zipped_layer)
-  return unzipped_layer, _sha256(ctx, unzipped_layer)
+def _layer_pair(ctx, layer):
+  zipped = False
+  for filetype in tgz_filetype:
```

Comment:
Can you hoist a function for each of `is_tgz` and `is_tar` (I think it will read a bit easier)

I'd kind of like to check that it matches one or the other and fail if neither instead of silently doing the wrong thing below.

Reply:
Done. Although the check should also be enforced by allow_files in the rule.

Note that tar filetypes are a superset of tgz filetypes.